addons/account/models/account_move.py

```diff
@@ -4253,49 +4253,49 @@ def get_account_notification(partner, moves, is_success):
             ]
 
         limit = job_count + 1
-        to_process = self.env['account.move']._read_group(
+        to_process = self.env['account.move'].search(
             [('send_and_print_values', '!=', False)],
-            groupby=['company_id'],
-            aggregates=['id:recordset'],
             limit=limit,
+            order='id asc',
         )
         need_retrigger = len(to_process) > job_count
         if not to_process:
             return
 
-        for _company, moves in to_process[:job_count]:
-            try:
-                # Lock moves
-                with self.env.cr.savepoint(flush=False):
-                    self._cr.execute('SELECT * FROM account_move WHERE id IN %s FOR UPDATE NOWAIT', [tuple(moves.ids)])
+        moves = to_process[:job_count]
 
-            except OperationalError as e:
-                if e.pgcode == '55P03':
-                    _logger.debug('Another transaction already locked documents rows. Cannot process documents.')
-                else:
-                    raise
+        try:
+            # Lock moves
+            with self.env.cr.savepoint(flush=False):
+                self._cr.execute('SELECT * FROM account_move WHERE id IN %s FOR UPDATE NOWAIT', [tuple(moves.ids)])
 
-            # Retrieve res.partner that executed the Send & Print wizard
-            sp_partner_ids = set(moves.mapped(lambda move: move.send_and_print_values.get('sp_partner_id')))
-            sp_partners = self.env['res.partner'].browse(sp_partner_ids)
-            moves_map = {
-                partner: moves.filtered(lambda m: m.send_and_print_values['sp_partner_id'] == partner.id)
-                for partner in sp_partners
-            }
+        except OperationalError as e:
+            if e.pgcode == '55P03':
+                _logger.debug('Another transaction already locked documents rows. Cannot process documents.')
```

Comment:
Indeed, there's a continue that is missing to avoid the process in case of already locked moves.

While I agree with the underlying problem that can end up with the cron timing out, there was a reason to groupby company, I contacted ryce to discuss a solution.